src/test/java/uk/nhs/hee/trainee/details/service/RabbitPublishServiceTest.java

```diff
@@ -121,10 +129,75 @@ void shouldPublishCojSignedEventWithUuid() {
         CojSignedEvent.class);
 
-    verify(rabbitTemplate).convertAndSend(any(), any(), eventCaptor.capture());
+    verify(rabbitTemplate).convertAndSend(any(), any(), eventCaptor.capture(),
+        any(CorrelationData.class));
 
     CojSignedEvent event = eventCaptor.getValue();
     assertThat("Unexpected programme membership ID.",
         event.getProgrammeMembershipTisId(), is(uuid.toString()));
   }
+
+  @Test
+  void shouldPublishCojSignedEventWithPmCorrelationId() {
+    ProgrammeMembership programmeMembership = new ProgrammeMembership();
+    programmeMembership.setTisId("123,456,7890");
+    Instant signedAt = Instant.now();
+    ConditionsOfJoining conditionsOfJoining
+        = new ConditionsOfJoining(signedAt, GoldGuideVersion.GG9);
+    programmeMembership.setConditionsOfJoining(conditionsOfJoining);
+
+    rabbitPublishService.publishCojSignedEvent(programmeMembership);
+
+    ArgumentCaptor<CorrelationData> correlationCaptor = ArgumentCaptor.forClass(
+        CorrelationData.class);
+
+    verify(rabbitTemplate).convertAndSend(any(), any(), any(CojSignedEvent.class),
+        correlationCaptor.capture());
+
+    CorrelationData correlation = correlationCaptor.getValue();
+    assertThat("Unexpected correlation ID.", correlation.getId(), is("123,456,7890"));
+  }
+
+  @ParameterizedTest()
+  @ValueSource(booleans = {true, false})
+  void shouldRemoveCallbackedMessageFromOutstandingConfirms(boolean isAck) {
+    ProgrammeMembership programmeMembership = new ProgrammeMembership();
+    programmeMembership.setTisId("123");
+    rabbitPublishService.outstandingConfirms.clear();
+    rabbitPublishService.outstandingConfirms.put(programmeMembership.getTisId(),
+        programmeMembership);
+    CorrelationData correlationData = new CorrelationData("123");
+
+    rabbitPublishService.handleRabbitAcknowledgement(isAck, correlationData);
+
+    assertThat("Unexpected outstanding confirms",
+        rabbitPublishService.outstandingConfirms.size(), is(0));
+  }
+
+  @Test
+  void shouldSetRabitConfirmCallback() {
+    rabbitPublishService.postConstruct();
+
+    verify(rabbitTemplate).setConfirmCallback(any());
+  }
+
+  @Test
+  void shouldNotThrowExceptionIfRabbitThrowsException() {
+    doThrow(new AmqpException("rabbit failed"))
+        .when(rabbitTemplate)
+        .convertAndSend(any(), any(), any(CojSignedEvent.class), any(CorrelationData.class));
+
+    ProgrammeMembership programmeMembership = new ProgrammeMembership();
+    programmeMembership.setTisId("123");
+    Instant signedAt = Instant.now();
+    ConditionsOfJoining conditionsOfJoining
+        = new ConditionsOfJoining(signedAt, GoldGuideVersion.GG9);
+    programmeMembership.setConditionsOfJoining(conditionsOfJoining);
+
+    try {
+      rabbitPublishService.publishCojSignedEvent(programmeMembership);
+    } catch (Exception e) {
+      fail("Unexpected exception when Rabbit not available");
+    }
```

Comment:
A cleaner approach would be to use `assertDoesNotThrow(() -> rabbitPublishService.publishCojSignedEvent(programmeMembership);, "Unexpected exception when Rabbit not available");`